src/Acuminator/Acuminator.Analyzers/StaticAnalysis/ActionHandlerAttributes/ActionHandlerAttributesFix.cs

```diff
@@ -0,0 +1,133 @@
+﻿using Acuminator.Utilities.Roslyn.Semantic;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Text;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Acuminator.Analyzers.StaticAnalysis.ActionHandlerAttributes
+{
+    internal enum FixOption
+    {
+        AddPXButtonAttribute,
+        AddPXUIFieldAttribute,
+        AddBothAttributes
+    }
+
+    [ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+    public class ActionHandlerAttributesFix : CodeFixProvider
+    {
+        internal const string FixOptionKey = nameof(FixOptionKey);
+
+        public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+            ImmutableArray.Create(Descriptors.PX1092_MissingAttributesOnActionHandler.Id);
+
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            var diagnostic = context.Diagnostics
+                .FirstOrDefault(d => d.Id.Equals(Descriptors.PX1092_MissingAttributesOnActionHandler.Id));
+
+            if (diagnostic == null || diagnostic.Properties == null)
+            {
+                return;
+            }
+
+            if (!diagnostic.Properties.TryGetValue(FixOptionKey, out string value))
+            {
+                return;
+            }
+
+            if (!Enum.TryParse(value, out FixOption option))
+            {
+                return;
+            }
+
+            var codeActionName = nameof(Resources.PX1092Fix).GetLocalized().ToString();
+            var codeAction = CodeAction.Create(
+                codeActionName,
+                cancellation => FixActionHandlerAttributes(context.Document, context.Span, option, cancellation));
+
+            context.RegisterCodeFix(codeAction, context.Diagnostics);
+        }
+
+        private async Task<Document> FixActionHandlerAttributes(Document document, TextSpan span, FixOption option, CancellationToken cancellation)
+        {
+            if (cancellation.IsCancellationRequested)
```

Comment:
@metallmonster shouldn't we use ThrowIfCancellationRequested?

Reply:
ok